tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -918,11 +920,71 @@ public function it_should_create_mime_types_for_allowed_sizes_only_via_global_va
 		$this->assertImageNotHasSizeSource( $attachment_id, 'not_allowed_size_200x150', 'image/webp' );
 	}
 
+	/**
+	 * Test image quality for image conversion.
+	 *
+	 * @test
+	 */
+	public function it_should_set_quality_with_image_conversion() {
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png', array( 'mime_type' => 'image/png' ) );
+
+		// Quality setting for the source image. For PNG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for PNG is 82.' );
+
+		// Temporary file.
+		$file = wp_tempnam();
+
+		// A PNG image will be converted to WebP whose quality should be 82 universally.
+		$editor->save( $file, 'image/webp' );
```

Comment:
`$file` can be null here and save will generate a filename, returning the saved file which you can then use for deleting.

Reply:
@adamsilverstein `$file` should always be a string here since `wp_tempnam()` always returns a string.